src/Targets/Microsoft.CSharp.DesignTime.targets

```diff
@@ -59,4 +112,88 @@
+                               IncludeFrameworkReferences="$(IncludeFrameworkReferencesFromNuGet)"
+                               NuGetPackagesDirectory="$(NuGetPackagesDirectory)"
+                               RuntimeIdentifier="$(_NuGetRuntimeIdentifierWithoutAot)"
+                               ProjectLanguage="$(Language)"
+                               ProjectLockFile="$(ProjectLockFile)"
+                               ProjectReferencesCreatingPackages="@(ProjectReferenceCreatingPackage)"
+                               ContentPreprocessorValues="@(NuGetPreprocessorValue)"
+                               ContentPreprocessorOutputDirectory="$(IntermediateOutputPath)\NuGet"
+                               TargetMonikers="$(NuGetTargetMoniker);$(_NuGetTargetFallbackMoniker)">
+
+      <Output TaskParameter="ResolvedAnalyzers" ItemName="_Analyzer" />
+      <Output TaskParameter="ResolvedCopyLocalItems" ItemName="_ReferenceCopyLocalPaths" />
+      <Output TaskParameter="ResolvedReferences" ItemName="_ReferencesFromNuGetPackages" />
+      <Output TaskParameter="ReferencedPackages" ItemName="_MyReferencedNuGetPackages" />
+      <Output TaskParameter="ContentItems" ItemName="_NuGetContentItems" />
+      <Output TaskParameter="FileWrites" ItemName="FileWrites" />
+    </ResolveNuGetPackageAssets>
+       
+    <ItemGroup>
+      <Packages Include="@(_MyReferencedNuGetPackages)"/>      
+    </ItemGroup>
+  </Target>
+  
+  <Target Name="RunLockFileToMSBuild">
+    <Message Text="     -> (Raising Lock File: ProjectLockFile=$(ProjectLockFile))" Importance="high"/>
+
+    <LockFileToMSBuild2 
+      ProjectLockFile="$(ProjectLockFile)">
+
+      <Output TaskParameter="TargetDefinitions" ItemName="TargetDefinitions" />
+      <Output TaskParameter="PackageDefinitions" ItemName="PackageDefinitions" />
+      <Output TaskParameter="FileDefinitions" ItemName="FileDefinitions" />
+      <Output TaskParameter="PackageDependencies" ItemName="PackageDependencies" />
+      <Output TaskParameter="FileDependencies" ItemName="FileDependencies" />
+    </LockFileToMSBuild2>
+
+  </Target>
+
+  <Target
+        Name="ResolveDependenciesDesignTime"
+        Returns="@(Packages)"
+        DependsOnTargets="ResolveAssemblyReferencesDesignTime;RunLockFileToMSBuild"
+        Condition="exists('$(ProjectLockFile)')">
+
+    <!--<Message Text="%(FileDefinitions.Identity)" Importance="high" />-->
+    
+    <GetDependenciesData TargetDefinitions="@(TargetDefinitions)"
+                               PackageDefinitions="@(PackageDefinitions)"
+                               FileDefinitions="@(FileDefinitions)"
+                               PackageDependencies="@(PackageDependencies)"
+                               FileDependencies="@(FileDependencies)">
+
+      <Output TaskParameter="DependenciesWorld" ItemName="_DependenciesWorld" />
+    </GetDependenciesData>
+   
+    <ItemGroup>
+      <Packages Include="@(_DependenciesWorld)"/>      
```

Comment:
>Packages [](start = 7, length = 8)

Doesn't this have to be PackageReference? The xaml file specifies the ItemType as PackageReference - don't they have to match?